### PR TITLE
Option to keep elements hidden, options page

### PR DESCRIPTION
**Pull request by duart38**
## things to keep in mind

1.this new feature is disabled by default. right click and go to the new options page to enable it and then click save

**option page also has an area to remove saved pages**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +28,19 @@ MIT: http://rhardih.mit-license.org
...
+- made the code smaller
+- fixed issue with ekill breaking pages like youtube or facebook
+- moved from replacing the whole HTML document to the built in remove()
+
+**1.5**
+
+- Added an options page 
+- made it possible to store elements a user has removed
+- added permission to access chrome.storage
+
```
Comment (rhardih) [inline README.md]: I think we'll just keep everything under a new 1.5 release, but I'll just fix that in the squash/merge process.

File: ekill.js [hunk 2]
```diff
@@ -1,37 +1,125 @@
...
+
+      let elementDump = document.getElementsByTagName(ekillStorage[i].element); // get a list of all the elements in this HTML page that is the same as the tag stored here e.g: <span> <div> etc tc
+      //console.log(elementDump);
+      for (let elem = 0; elem < elementDump.length; elem++) { // elem here is an html element
+        //console.log(elementDump[elem])
+        if (elementDump[elem].outerHTML === ekillStorage[i].outerHTML) { // cross checking if the outerHTMLs are the same between the stored value and the current itteration value
+          elementDump[elem].remove(); // remove the element.
+          break; // break out of the loop if we found the element to keep that CPU nice and cold..
+        }
+      }
```
[the remark this answers is not in the document]
Comment (duart38) [reply ekill.js]: do you have any ideas here? i was thinking about the id attribute but i know allot of front-end devs that barely use id's  (kind of guilty of this myself) sometimes even ignoring classes all together and adding for example an empty span with some text in it.

**Comment by duart38** [comment]
## ok so i:
1. changed some code to make it work with firefox. 
2. also changed the settings page to be a bit nicer.
3. tried cleaning up the code
4. added a delayed function to handle elements that have not yet been added to the DOM

**Comment by maticrivo** [comment]
Just a thought, haven't looked at your code yet..
Maybe instead of doing a delayed function you can take advantage of `MutationObserver` and observe the DOM to see when the element you want to kill is added to the DOM

File: ekill.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhardih) [reply ekill.js]: I think it needs to be a mix of different things in order to provide a reliable solution, but I'll have to play around with some different approaches. This is a good enough start. 👍

**Comment by rhardih** [comment]
Cool @duart38 👍 

I'll merge and do some experimentation on the grudge branch and ping you in the PR from grudge -> master.

**Pull request merged**